cmd/kubeadm/app/cmd/join.go

```diff
@@ -203,6 +203,12 @@ func NewCmdJoin(out io.Writer, joinOptions *joinOptions) *cobra.Command {
 			err = data.Run()
 			kubeadmutil.CheckErr(err)
 		},
+		// This allows for `kubeadm join <master>`. Ideally we should encourage to
```

Comment:
Sorry but I don't agree on this comment. Please remove it.
If we want to define a guideline this is not the right place; nevertheless, shifting from arguments to flags is really opinionated; instead IMO this should be decided use case by use case.
In this case phasee is a valid host name, so this makes args validation difficult, but your join process will fail before starting/during discovery.  We should make sure the error message is clear to the users.

Reply:
> If we want to define a guideline this is not the right place; nevertheless, shifting from arguments to flags is really opinionated; instead IMO this should be decided use case by use case.

You are absolutely right. Sorry.